service/history/callback_queue_active_task_executor.go

```diff
@@ -0,0 +1,257 @@
+	workflowCache wcache.Cache,
+	logger log.Logger,
+	metricsHandler metrics.Handler,
+	config *configs.Config,
+	callerGetter func(namespaceID, destination string) HTTPCaller,
+) *callbackQueueActiveTaskExecutor {
+	return &callbackQueueActiveTaskExecutor{
+		taskExecutor: taskExecutor{
+			shardContext:   shardCtx,
+			cache:          workflowCache,
+			logger:         logger,
+			metricsHandler: metricsHandler.WithTags(metrics.OperationTag(metrics.OperationCallbackQueueProcessorScope)),
+		},
+		config:       config,
+		clusterName:  shardCtx.GetClusterMetadata().GetCurrentClusterName(),
+		callerGetter: callerGetter,
+	}
+}
+
+func (t *callbackQueueActiveTaskExecutor) Execute(
+	ctx context.Context,
+	executable queues.Executable,
+) queues.ExecuteResponse {
+	task := executable.GetTask()
+	taskType := "Active" + task.GetType().String()
+
+	namespaceTag, replicationState := getNamespaceTagAndReplicationStateByID(
+		t.shardContext.GetNamespaceRegistry(),
+		task.GetNamespaceID(),
+	)
+	metricsTags := []metrics.Tag{
+		namespaceTag,
+		metrics.TaskTypeTag(taskType),
+		metrics.OperationTag(taskType),
+	}
+
+	// Check namespace handover state once when processing is started. Callback tasks may take up to 10 seconds
+	// by default but we avoid checking again before committing the result to complete inflight tasks even if
+	// the namespace transitions to handover state during processing.
+	if replicationState == enumspb.REPLICATION_STATE_HANDOVER {
+		// TODO: Move this logic to queues.Executable when metrics tags don't need to be returned from task executor.
+		return queues.ExecuteResponse{
+			ExecutionMetricTags: metricsTags,
+			ExecutedAsActive:    true,
+			ExecutionErr:        consts.ErrNamespaceHandover,
+		}
+	}
+
+	var err error
+	switch task := task.(type) {
+	case *tasks.CallbackTask:
+		err = t.processCallbackTask(ctx, task)
+	default:
+		err = queues.NewUnprocessableTaskError("unknown task type")
+	}
+
+	return queues.ExecuteResponse{
+		ExecutionMetricTags: metricsTags,
+		ExecutedAsActive:    true,
+		ExecutionErr:        err,
+	}
+}
+
+func (t *callbackQueueActiveTaskExecutor) processCallbackTask(ctx context.Context, task *tasks.CallbackTask) error {
+	ctx, cancel := context.WithTimeout(ctx, t.config.CallbackTaskTimeout())
+	defer cancel()
+
+	_, release, ms, err := t.getValidatedMutableStateForTask(
+		ctx, task, func(ms workflow.MutableState) error {
+			return t.validateCallbackTask(ms, task)
+		},
+	)
+	if err != nil {
+		return err
+	}
+	// We're just reading mutable state here, no need to unload from cache.
+	defer release(nil)
+
+	// Callback is validated above and must be present in map.
+	callback := ms.GetExecutionInfo().Callbacks[task.CallbackID]
+	switch variant := callback.PublicInfo.GetCallback().GetVariant().(type) {
+	case *commonpb.Callback_Nexus_:
+		completion, err := t.getNexusCompletion(ctx, ms)
+		// We got what we need from mutable state, release now and reaqcuire later when processing is done.
```

Comment:
We got what we need from mutable state; release now and reacquire later when processing is done.

(comma splice and typo in reacquire)